docusaurus/docs/Android/04-compose/04-message-components/07-selected-reactions-menu.mdx

```diff
@@ -0,0 +1,176 @@
+
+```kotlin
+if (selectedMessageState is SelectedMessageReactionsState) {
+    SelectedReactionsMenu(
+        ..., // State
+        onMessageAction = { action ->
+            composerViewModel.onMessageAction(action)
+			listViewModel.onMessageAction(action)
+        },
+        onDismiss = {
+			listViewModel.removeOverlay()
+        }
+    )
+}
+```
+
+In the snippet above, you propagate the `action` to the `composerViewModel` and `listViewModel`, for them to store the latest action. This will update the UI accordingly.
+
+Alternatively, you call `listViewModel.removeOverlay()` to remove the overlay from the screen, in `onDismiss()`. It's important to note that `onMessageAction()` calls `removeOverlay()` internally, to hide the overlay.
+
+Next, let's see how to customize the overlay.
+
+## Customization
+
+You can customize the reactions you show:
+
+```kotlin
+@Composable
+fun SelectedReactionsMenu(
+    reactionTypes: Map<String, Int> = ChatTheme.reactionTypes,
+    ... // Message and actions
+)
+```
+
+* `reactionTypes`: Allows you customize reactions shown in the overlay. By default it uses `ChatTheme.reactionTypes` that is exposed through the [`ChatTheme`](../02-general-customization/01-chat-theme.mdx) component.
+
+If you want to customize the reactions, you can set up custom reactions, using the [`ChatTheme`](../02-general-customization/01-chat-theme.mdx)'s `reactionTypes` parameter.
+
+By default `SelectedReactionsMenu` looks like a bottom sheet, however you can customize it to look like a completely different component, such as a dialog, a drawer or whatever helps you retain the look and feel of your app.
+
+```kotlin
+if (selectedMessageState is SelectedMessageReactionsState) {
+    // Use a Modifier to customize the appearance
+    SelectedReactionsMenu(
+        // Use a Modifier to customize the appearance
+        modifier = Modifier
+            .align(Alignment.Center)
+            .padding(horizontal = 20.dp)
+            .wrapContentSize(),
+        // Assign a different shape to the Composable element
+        shape = ChatTheme.shapes.attachment,
+        currentUser = user,
+        onMessageAction = { action ->
+            // Handle message action
+        },
+        onDismiss = { 
+            // Handle dismiss 
+        },
+        ...
+    )
+}
+```
+
+The code above will produce the following UI:
+
+![Custom SelectedReactionsMenu component](../../assets/compose_custom_selected_reactions_menu_shape_and_alignment.png)
+
+`SelectedReactionsMenu` provides you with Composable slots that are ready for more extensive customizations.
+
+```kotlin
+@Composable
+fun SelectedReactionsMenu(
+    ...,
+    headerContent: @Composable ColumnScope.() -> Unit = {
+        // Header content
+    },
+    centerContent: @Composable ColumnScope.() -> Unit = {
+        // Center content
+    }
+)
+```
+
+* `headerContent`: Allows you to customize the content shown at the top of the menu. By default it shows `ReactionOptions`.
```

Comment:
I think our newer unofficial guidelines are to avoid mentioning exact implementations like these since they are inside of `DefaultX`. 

Using "reactions options" instead of `ReactionOptions` might be advisable.

Please double check this with @filbabic

Reply:
Agreed